src/main/java/com/theelixrlabs/healthcare/service/PatchPatientService.java

```diff
@@ -0,0 +1,80 @@
+    private final MessageUtil messageUtil;
+    private final DoctorRepository doctorRepository;
+
+    public PatchPatientService(PatientRepository patientRepository, Validator validator, MessageUtil messageUtil, DoctorRepository doctorRepository) {
+        this.patientRepository = patientRepository;
+        this.validator = validator;
+        this.messageUtil = messageUtil;
+        this.doctorRepository = doctorRepository;
+    }
+
+    /**
+     * Replace existing patient details based on request.
+     *
+     * @param patientId  Patient unique UUID
+     * @param patientDto Data transfer object containing patient information.
+     * @return The replaced patient details.
+     * @throws CustomException If validation fails or patient already exists.
+     */
+    public PatientDto patchPatientById(String patientId, PatientDto patientDto) throws CustomException {
+        UUID validPatientId = validator.validateAndConvertToUUID(patientId, MessageConstants.INVALID_UUID);
+        validator.validatePatchPatient(patientDto);
+        Optional<PatientModel> optionalPatient = patientRepository.findById(validPatientId);
+        if (optionalPatient.isEmpty()) {
+            throw new ResourceNotFoundException(messageUtil.getMessage(PatientConstants.PATIENT_NOT_FOUND_KEY));
+        }
+        PatientModel existingPatient = optionalPatient.get();
+        if (!StringUtils.isBlank(patientDto.getPatientAadhaarNumber()) && !patientDto.getPatientAadhaarNumber().equals(existingPatient.getPatientAadhaarNumber())) {
+            String aadhaarNumber = patientDto.getPatientAadhaarNumber();
+            String formattedAadhaarNumber = formatAadhaarNumber(aadhaarNumber);
+            if (patientRepository.findByPatientAadhaarNumber(formattedAadhaarNumber).isPresent() || doctorRepository.findByAadhaarNumber(formattedAadhaarNumber).isPresent()) {
+                throw new CustomException(messageUtil.getMessage(MessageConstants.PATIENT_AADHAAR_NUMBER_EXISTS));
+            }
+            existingPatient.setPatientAadhaarNumber(formattedAadhaarNumber);
+        }
+        if (isFieldNullOrUnchanged(patientDto.getPatientFirstName(), existingPatient.getPatientFirstName())) {
+            existingPatient.setPatientFirstName(patientDto.getPatientFirstName());
+        }
+        if (isFieldNullOrUnchanged(patientDto.getPatientLastName(), existingPatient.getPatientLastName())) {
+            existingPatient.setPatientLastName(patientDto.getPatientLastName());
+        }
+        patientRepository.save(existingPatient);
+        return PatientDto.builder()
+                .id(existingPatient.getId())
+                .patientFirstName(existingPatient.getPatientFirstName())
+                .patientLastName(existingPatient.getPatientLastName())
+                .patientAadhaarNumber(existingPatient.getPatientAadhaarNumber())
+                .build();
+    }
+
+    private boolean isFieldNullOrUnchanged(String newValue, String existingValue) {
```

Comment:
Is the method name is correct ? isFieldNullOrUnchanged .  
isFieldNotNullAndHasNewValue or simply name it as hasValueForUpdate

Reply:
changed to hasValueForUpdate()